enterprise/server/backends/authdb/authdb.go

```diff
@@ -429,32 +430,46 @@ func (d *AuthDB) GetAPIKeyGroupFromAPIKeyID(ctx context.Context, apiKeyID string
 }
 
 func (d *AuthDB) LookupUserFromSubID(ctx context.Context, subID string) (*tables.User, error) {
-	user := &tables.User{}
-	// TODO(zoey): switch this to JOIN
-	err := d.h.TransactionWithOptions(ctx, db.Opts().WithStaleReads(), func(tx interfaces.DB) error {
-		rq := tx.NewQuery(ctx, "authdb_lookup_user_from_subid").Raw(
```

Comment:
Is there a practical situation where having these two selects outside of a single transaction would cause problems? Seems that would be simpler than trying to combine everything into a single complex query.

Reply:
Is this that much more complicated? The second query is already doing an implicit inner join of `Groups` and `UserGroups`; this is just switching that to a left join of the column of `Users` that matches the `sub_id`, `UserGroups`, and `Groups`. It should be about as fast and be only one query as opposed to multiple (which reduces our overall qps, though admittedly not by that much as this is by far not the bulk of the queries we make).

I agree that removing the transaction here without merging the queries is likely safe, given that the user returned by the initial query is unlikely to change between the two queries, if it exists, and if it doesn't exist the second query will never run. I think the one potential weird failure mode is if a user was deleted between the two queries, we would get back a user that was a member of no groups, which would be a weird, inconsistent state of that user that never should have existed, but it would be unlikely to actually cause any problems, since it would just mean that user would not authenticate as a member of any groups.

All of that said, this is hardly a critical change, it just felt like low-hanging fruit to consolidate queries that didn't need to be a series of separate queries. If you think it's an issue, I can just close the PR.

Reply:
I think it wouldn't look as complicated if we made the index on sub_id unique at which point it would become 3 plain joins. There's no reason for there to be multiple users with the same sub_id. 

The other complication is that you need additional logic to differentiate the case of "no user" and "user and no groups" due to having the data returned in a single query. 

This is not a performance sensitive code path so I don't see the incentive to try to force everything into a single query.

Reply:
Works for me, thanks for your input!

I might open an issue on unique-ifying the sub_id